build/build.go

```diff
@@ -591,25 +592,79 @@ func (s *Session) buildImportPathWithSrcDir(path string, srcDir string) (*Packag
 	return pkg, archive, nil
 }
 
+const (
+	hashDebug = false
+)
+
+var (
+	gopherJsBinaryHash string
+)
+
+func hashGopherJsBinary() (string, error) {
+	if gopherJsBinaryHash != "" {
+		return gopherJsBinaryHash, nil
+	}
+
+	binHash := sha256.New()
+	binPath, err := os.Executable()
+	if err != nil {
+		return "", fmt.Errorf("could not locate GopherJS binary: %v", err)
+	}
+	binFile, err := os.Open(binPath)
+	if err != nil {
+		return "", fmt.Errorf("could not open %v: %v", binPath, err)
+	}
+	defer binFile.Close()
+	if _, err := io.Copy(binHash, binFile); err != nil {
+		return "", fmt.Errorf("failed to hash %v: %v", binPath, err)
+	}
+	gopherJsBinaryHash = fmt.Sprintf("%#x", binHash.Sum(nil))
+	return gopherJsBinaryHash, nil
+}
+
 func (s *Session) BuildPackage(pkg *PackageData) (*compiler.Archive, error) {
 	if archive, ok := s.Archives[pkg.ImportPath]; ok {
 		return archive, nil
 	}
 
+	// For non-main and test packages we build up a hash that will help
+	// determine staleness. Set hashDebug to see this in action. The format is:
+	//
+	// ## sync
```

Comment:
Why do we need this?

Reply:
This is actually a package import path for `sync` - it was an example. I've changed the example now to use placeholders instead of concrete examples, to hopefully make it clearer.